package/MDAnalysis/transformations/translate.py

```diff
@@ -112,14 +124,14 @@ def center_in_box(ag, center='geometry', point=None, wrap=False):
         if point.shape != (3, ) and point.shape != (1, 3):
             raise ValueError('{} is not a valid point'.format(point))
     try:
-        if center == 'geometry':
+        if center_of == 'geometry':
```

Comment:
why do we have center_of here all the time? Can't you make use of the generic [`center`](https://www.mdanalysis.org/docs/documentation_pages/core/groups.html?highlight=center#MDAnalysis.core.groups.AtomGroup.center) as well? it allows to center with respect to arbitrary weights. People *will* want to do this. Think of IDP proteins where you want to calculate a center with respect to the flexibility of residues.

Reply:
You could have an argument `weights='mass'` like we have for the encore methods. The argument would accept `mass` and `centroid` as string arguments or an array-like object or None.